src/Jss.js

```diff
@@ -45,9 +45,19 @@ export default class Jss {
    * Create a Style Sheet.
    */
   createStyleSheet(styles: Object, options: StyleSheetOptions): StyleSheet {
+    let generateClassName
+    if (options
+      && options.sheetsRegistryContext
+      && this.options.generateClassName === generateClassNameDefault) {
```

Comment:
Thats the problem, what if user changed the generateClassName algorithm and now he uses the ThemeProvider with registry?